src/Illuminate/Auth/SessionGuard.php

```diff
@@ -129,10 +129,8 @@ public function user()
         // request, and if one exists, attempt to retrieve the user using that.
         $user = null;
```

Comment:
I think is safe to remove this `$user` since `retrieveById` will either be null or the value we need.

Reply:
@jycr753 no it can not be removed since `if` statements in PHP are [short-circuited](https://stackoverflow.com/a/2535578/1564365). This it could render `PHP error:  Undefined variable: user on line X`